src/main/java/org/htwkvisu/org/pois/BasicPOI.java

```diff
@@ -3,17 +3,44 @@
 import javafx.geometry.Point2D;
 import javafx.scene.canvas.GraphicsContext;
 import javafx.scene.paint.Color;
+import org.hibernate.mapping.Array;
 import org.htwkvisu.gui.MapCanvas;
 import org.htwkvisu.org.IMapDrawable;
 
 public class BasicPOI implements IMapDrawable {
     private final ScoreType type;
-    private Point2D position = new Point2D(0, 0);
-    private static final double POINT_SIZE = 3;
+    private final Point2D position;
+    private static final double POINT_SIZE = 2;
+    private final Color color;
 
     public BasicPOI(ScoreType type, Point2D position) {
         this.type = type;
         this.position = position;
+        this.color = calcTypeColor();
+    }
+
+    private Color calcTypeColor() {
+        Color tmpColor = Color.BLACK;
+        for (Category cat : Category.values()) {
+            for (ScoreType scoreType : cat.getTypes()) {
+                if (type == scoreType) {
+                    tmpColor = getColorForCategory(cat);
+                }
+            }
+        }
+        return tmpColor;
+    }
+
+    private Color getColorForCategory(Category cat) {
+        Color tmpColor;
```

Comment:
Try to add a field `Color color` to the `Category` as first param of the constructor. Add furthermore getters and setters from this new field. Then you can call easily `category.getColor();`.